javagen/src/main/java/com/azure/autorest/mapper/ProxyMethodMapper.java

```diff
@@ -251,6 +268,23 @@ public Map<Request, List<ProxyMethod>> map(Operation operation) {
                 proxyMethodParameters.add(i, binaryDataParam);
                 builder.parameters(proxyMethodParameters);
                 proxyMethods.add(builder.build());
+
+                if(asyncRestResponseReturnType instanceof GenericType
+                        && ((GenericType) asyncRestResponseReturnType).getTypeArguments()[0] instanceof GenericType
+                        && ((GenericType) ((GenericType) asyncRestResponseReturnType).getTypeArguments()[0]).getName().equals("ResponseBase")) {
+                    IType asyncResponseWithNoHeaders = getAsyncRestResponseReturnType(operation, responseBodyType,
+                            settings.isDataPlaneClient(), settings, true);
+                    builder.returnType(asyncResponseWithNoHeaders);
+                    builder.name(operationName + "NoCustomHeaders");
+                    builder.customHeaderIgnored(true);
+
+                    proxyMethods.add(builder.build());
+
+                    // reset builder state
+                    builder.returnType(asyncRestResponseReturnType);
+                    builder.name(operationName);
+                    builder.customHeaderIgnored(false);
+                }
```

Comment:
If the code is same as above (line 235), it better to be extracted to a function (or lambda if that's easier).